set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java

```diff
@@ -65,6 +63,16 @@ private UserSetQuery buildSearchQuery(Map<String, String> searchCriteria, String
 	    }
 	}
 
+	if (searchCriteria.containsKey(WebUserSetFields.SET_ID)) {
+		String setId = searchCriteria.get(WebUserSetFields.SET_ID);
+		if (setId != null && ! UserSetUtils.isSetIdNumeric(setId)) {
```

Comment:
the not null check is redundant Integer.parseInt already checks it

Reply:
I will remove it